Applet/src/com/android/javacard/keymaster/RemotelyProvisionedComponentDevice.java

```diff
@@ -289,9 +292,14 @@ public void constructPartialSignedData(byte[] scratchPad, short coseKeysCount,
     short keysToSignLen = (short) (coseKeysArrHeaderLen + totalCoseKeysLen);
 
     // Calculate the payload array header len
-    short paylaodArrHeaderLen = 1; // Array of 5 elements occupies 1 byte.
-    short payloadLen = (short) (paylaodArrHeaderLen + versionLength + certTypeLen 
-        + deviceInfoLen + challengeHeaderLen + challengeLen + keysToSignLen);
+    short paylaodArrHeaderLen = 1; // Array of 2 elements occupies 1 byte.
+    short csrPaylaodArrHeaderLen = 1; // Array of 4 elements occupies 1 byte.
```

Comment:
We may add some comments here so that it will be easy for readers to understand.